### PR TITLE
[HOPSWORKS-2195] Attach Jupyter configuration to notebook as Xattr

### PR DESCRIPTION
**Pull request by gibchikafa**

### CODE REVIEW

**Review by robzor92 — APPROVED** [review]

File: hdfscontents/hdfsmanager.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
 import mimetypes
 import nbformat
 from traitlets import Instance, Integer, Unicode, default
+from hops import xattr
```
Comment (robzor92) [inline hdfscontents/hdfsmanager.py]: As we want hdfscontents to work in environments where hops is not installed, can you add a try here, and log the exception?

File: hdfscontents/hdfsmanager.py [hunk 2]
```diff
@@ -375,16 +380,37 @@ def rename_file(self, old_path, new_path):
...
         except Exception as e:
             raise web.HTTPError(500, u'Unknown error renaming file: %s %s' % (old_path, e))
 
     def info_string(self):
         return "Serving notebooks from HDFS directory: %s" % self.root_dir
+
+    def get_notebook_jupyter_configuration_xatrr(self, hdfs_path):
+        self.log.debug("Getting jupyter configuration xattr for %s", hdfs_path)
+        jupyter_configuration = {}
+        if ".ipynb" not in hdfs_path:
```
Comment (robzor92) [inline hdfscontents/hdfsmanager.py]: Can you check if the xattr module is loaded? If it is not we should not call the get_xattr endpoint.

File: hdfscontents/hdfsmanager.py [hunk 3]
```diff
@@ -375,16 +380,37 @@ def rename_file(self, old_path, new_path):
...
+            return jupyter_configuration
+        try:
+            jupyter_configuration = xattr.get_xattr(hdfs_path, self.jupyter_configuration_xattr_name)
+        except Exception as e:
+            self.log.debug(u'Failed to get jupyter configuration xattr for %s %s', hdfs_path, e, exc_info=True)
+        return jupyter_configuration
+
+    def set_notebook_jupyter_configuration_xatrr(self, hdfs_path, jupyter_configuration):
```
Comment (robzor92) [inline hdfscontents/hdfsmanager.py:419]: Also check that xattr is loaded here too.

File: hdfscontents/hdfsmanager.py (hunk 2 above)
Comment (robzor92) [reply hdfscontents/hdfsmanager.py]: For reference you can check like this https://stackoverflow.com/questions/30483246/how-to-check-if-a-python-module-has-been-imported

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: hdfscontents/hdfsmanager.py [hunk 4]
```diff
@@ -17,6 +17,12 @@
 import mimetypes
 import nbformat
 from traitlets import Instance, Integer, Unicode, default
+try:
+    from hops import xattr
+except ImportError:
+    pass
+
+from hops.exceptions import RestAPIError
```
Comment (robzor92) [inline hdfscontents/hdfsmanager.py]: Can you also move this to being inside the try statement?

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: hdfscontents/hdfsmanager.py (hunk 4 above)
Comment (gibchikafa) [reply hdfscontents/hdfsmanager.py]: Thanks. It is also unused import. My bad.

File: hdfscontents/hdfsmanager.py (hunk 3 above)
Comment (gibchikafa) [reply hdfscontents/hdfsmanager.py:419]: When xattr is not laoded `get_notebook_jupyter_configuration_xatrr` will return an empty set. So there will be no problem here.

**Review by robzor92 — APPROVED** [review]

**Pull request merged**